iceoryx2-bb/cxx/include/iox2/container/expected.hpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#ifndef IOX2_INCLUDE_GUARD_CONTAINER_EXPECTED_HPP
+#define IOX2_INCLUDE_GUARD_CONTAINER_EXPECTED_HPP
+
+#include "iox2/legacy/expected.hpp"
+#include <type_traits>
+
+namespace iox2 {
+namespace container {
+
+struct InPlaceT { };
+// NOLINTNEXTLINE(readability-identifier-naming), for consistency with C++23 code using std::expected
+constexpr InPlaceT in_place {};
```

Comment:
This is analogous to `iox2::container::nullopt`. Shall we change all constants to upper camel case?

Reply:
Analogous to `optional`, if this is intended as a drop-in replacement for [the `std` version](https://en.cppreference.com/w/cpp/utility/in_place.html), it needs to remain lower case.

This same type would also be used for `optional` (and possibly other types), so it would move to a different header eventually.

Reply:
That's what we do when opting into using the standard facilities. The assumption was that the implementation we deploy on either won't have a `std::nullopt`, or won't have it as part of the safety-critical subset.